Add script to boot simulators using webkitpy.

bls1999 opened the pull request:
#### c261cf90ec3f34f85a7fc1ee3c898d82a76c17eb
<pre>
Add script to boot simulators using webkitpy.
<a href="https://rdar.apple.com/127062283">rdar://127062283</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=273260">https://bugs.webkit.org/show_bug.cgi?id=273260</a>

Reviewed by Jonathan Bedard.

This PR adds a script that is able to tie into `webkitpy.xcode.simulated_device.SimulatedDeviceManager`
to boot simulated devices.

* Tools/Scripts/boot-simulators: Added.
(init): Sets up script arguments and logging.
(main): Determines the appropriate port to use and generates/satisfies corresponding requests.

Canonical link: <a href="https://commits.webkit.org/278179@main">https://commits.webkit.org/278179@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d95b7cc9883251e1b65b87c4e1b322de302307a4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49741 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29029 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/52792 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/52983 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/417 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/52046 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/35050 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/26646 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/52983 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/51841 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/26557 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/52792 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/52983 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/23998 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/52792 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/8113 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/45908 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/52792 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/54565 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/24829 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/20985 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/54565 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/26095 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/52792 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/54565 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/26943 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7159 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/25820 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/boot-simulators:
```diff
@@ -0,0 +1,85 @@
+
+def main():
+    _log = logging.getLogger()
+    _log.setLevel(logging.DEBUG)
```
`webkitcorepy` has a `-v/-q` argument template that's hooked into the logging level. It's probably an easy usability win here, see `Tools/Scripts/libraries/webkitcorepy/webkitcorepy/arguments.py`

bls1999 replied:
Added in recent commit!

JonWBedard commented on Tools/Scripts/boot-simulators:
```diff
@@ -0,0 +1,88 @@
+def init():
+    parser = argparse.ArgumentParser(description='Boots a specified number of simulators of the default device type.')
+
+    # positional args
```
Comments like these (except for the `FIXME` one) aren't very helpful, and should usually be expressed by better variable names, if at all.

bls1999 replied:
Removed the comments in latest commit.

bls1999 committed:
Add script to boot simulators using webkitpy.
rdar://127062283
https://bugs.webkit.org/show_bug.cgi?id=273260

Reviewed by Jonathan Bedard.

This PR adds a script that is able to tie into `webkitpy.xcode.simulated_device.SimulatedDeviceManager`
to boot simulated devices.

* Tools/Scripts/boot-simulators: Added.
(init): Sets up script arguments and logging.
(main): Determines the appropriate port to use and generates/satisfies corresponding requests.

Canonical link: https://commits.webkit.org/278179@main
